File: data structures/Go/toLowerCase.go
package main

//O(N)
//O(1)
func toLowerCase(s string) string {

	if len(s) == 0 {
		return ""
	}

	result := ""

	for _, char := range s {
		if char >= 'A' && char <= 'Z' {
			char += 32
		}

		result += string(char)
	}

	return result
}

File: C++/toLowerCase.cpp
// Time: O(N)
// Spce: O(N)
#include <string>

class Solution
{
public:
    std::string toLowerCase(std::string s)
    {
        if (s.empty())
            return "";

        std::string result;

        for (char c : s)
        {
            if (c >= 'A' && c <= 'Z')
                c = c + 32;
            result += c;
        }

        return result;
    }
};